hcl/hcl.go

```diff
@@ -37,11 +37,26 @@ type Config struct {
 	Stack     *Stack
 }
 
+type GitConfig struct {
+	BaseRef              string // BaseRef is the general base git ref.
+	DefaultBranchBaseRef string // DefaultBranchBaseRef is the baseRef when in default branch.
+	Branch               string // Branch is the default branch.
+	Remote               string // Remote is the default remote.
+}
+
+type RootConfig struct {
+	Git GitConfig
+}
```

Comment:
we have specs for git config goodies: https://github.com/mineiros-io/terramate/blob/main/docs/project-config.md
we can update the specs if something changed here (I think some names did), or fix the code (I'm quite good with both).